test/test_c10d.py

```diff
@@ -1646,52 +1646,60 @@ def tearDown(self):
     def world_size(self):
         return 2
 
-    def _prepare_single_device_module(self, process_group, gpus, global_batch_size):
+    def _prepare_single_device_module(self, process_group, devices, device_ids, global_batch_size):
         model = Net()
         ddp_model = DistributedDataParallel(
-            copy.deepcopy(model).cuda(gpus[0]),
-            device_ids=gpus,
+            copy.deepcopy(model).to(devices[0]),
+            device_ids=device_ids,
```

Comment:
`device_ids` in DDP should be able to directly handle devices, as we wrote:

```python
device_ids (list of int or torch.device): CUDA devices. This should ...
```

Reply:
The existing tests exercised both and I didn't want to change that. We should test both the list of integers and list of `torch.device` cases in the suite (it could be broken out as a separate test).